packages/utils/src/lib/git.integration.test.ts

```diff
@@ -16,3 +24,66 @@ describe('getLatestCommit', () => {
     );
   });
 });
+
+describe('branchHasChanges', () => {
+  it('should return true if some changes are given', async () => {
+    await makeStatusDirty();
+    await expect(branchHasChanges()).resolves.toBe(true);
+    await makeStatusClean();
+  });
+  it('should return false if no changes are given', async () => {
+    await expect(branchHasChanges()).resolves.toBe(false);
+  });
+});
+
+describe('guardAgainstDirtyRepo', () => {
+  it('should throw if history is dirty', async () => {
+    await makeStatusDirty();
+    await expect(guardAgainstDirtyRepo()).rejects.toThrow(
+      'Repository should be clean before we you can proceed',
+    );
+    await makeStatusClean();
+  });
+  it('should not throw if history is clean', async () => {
+    await expect(guardAgainstDirtyRepo()).resolves.toEqual(void 0);
+  });
+});
+
+describe('getCurrentBranchOrTag', () => {
+  it('should log current branch', async () => {
+    await expect(getCurrentBranchOrTag()).resolves.toEqual(expect.any(String));
+  });
+});
+
+describe('safeCheckout', () => {
+  // eslint-disable-next-line vitest/require-hook
+  let initialBranch: string = '';
```

Comment:
This rule only triggers when one tries to instantiate the variable, not when it is only declared.
Seeing that it is assigned a value before execution of any tests, does the default value need to be there?

```suggestion
  let initialBranch: string;
```